### ISSUE TITLE
proposal(x/swaprouter): keep swap messages per module as opposed to unifying in x/swaprouter

### ISSUE THREAD

**Issue by p0mvn** [osmosis-labs/osmosis#3792]
## Background

Proposed by @AlpinYukseloglu 

From a client perspective, it might feel like a bad UX to have the swap messages for all pools be in `x/swaprouter` while pool creation messages be per module.

It is not intuitive that for creating a concentrated liquidity pool, we need to submit a message to `x/concentrated-liquidity`. On the other hand, to swap against a concentrated liquidity pool, we need to submit a message to `x/swaprouter`.

## Suggested Design

The proposal is to keep the core swap keeper logic in `x/swaprouter`. However, have the swap messages be defined per keeper. Each module's swap message calls into the `x/swaprouter` `RouteExactAmountIn/Out`.

Additionally, this would avoid the need for deprecating the existing swap messages in `x/gamm`.

There is still value in introducing `x/swaprouter`. It is reducing common boilerplate for pool creation and swapping. 

### Disadvantages

Fragmenting swap entry points for clients as opposed to having just one.

### Advantages

As a client, given a pool-related message, I do not have to think about whether to go through `x/swaprouter` or the pool's module (`x/gamm` or `x/concentrated-liquidity`). Instead, always go directly to the pool's underlying module for message creation.

## Acceptance Criteria
- the decision is made

**Comment by AlpinYukseloglu** [comment]
Tldr: turn `x/swaprouter` into a purely backend abstraction that serves the primary purpose of minimizing duplicate code. If a user wants to interact with concentrated liquidity or GAMM, all messages have to be sent to `x/concentrated-liquidity` or `x/gamm` respectively.

In practice, this would mean just adding  swap-related messages to `x/concentrated-liquidity` and routing them to `x/swaprouter` behind the scenes (as is currently done with GAMM).

The primary motivation for this is the fact (or assumption?) that only _part_ of concentrated liquidity & GAMM logic can be abstracted, meaning that integrators and developers are forced to understand the nuances of our architecture if they want to use concentrated liquidity (e.g. even if `CreatePool` is used relatively infrequently, someone integrating with Osmosis likely still has to support it, forcing them to have to break our abstractions)

**Comment by AlpinYukseloglu** [comment]
Looks like we're going with keeping swaps in `x/swaprouter`, not defining any concentrated-liquidity-side swap msgs, and keeping `CreatePool` in both

Closing this thread as complete!
